hydradx/model/amm/amm.py

```diff
@@ -2,6 +2,16 @@
 from ..amm import omnipool_amm as oamm
 
 
+def agent_dict(token_list: list = (), r_values: dict = None, s_values: dict = None, p_values: dict = None) -> dict:
+    if r_values is None:
+        r_values = {}
+    return {
+        'r': r_values or {token: 0 for token in token_list},
+        's': s_values or {},
```

Comment:
Yes, it's that. It's because the 'or' operator doesn't necessarily return a boolean value. Specifically, it returns the first non-false value on either side of the expression, and if both are false(y) it returns the second value. So True or False == True, and False or True == True, but also 1 or 2 == 1, 2 or 1 == 2, False or 0 == 0, and None or {} == {}. I believe I just got so used to it that I forgot it's confusing.

But I've changed this already and I plan to change it again. Ultimately I want to use the next layer to enforce that everyone is working from the same asset list, so we don't get things like KeyErrors when iterating through agents (which I have had to hack around already).

Reply:
Interesting, did not know that about Python's "or"!